bgpd/bgp_zebra.c

```diff
@@ -1722,11 +1727,7 @@ void bgp_zebra_announce_table_all_subtypes(struct bgp *bgp, afi_t afi,
 		return;
 
 	for (dest = bgp_table_top(table); dest; dest = bgp_route_next(dest))
-		for (pi = bgp_dest_get_bgp_path_info(dest); pi; pi = pi->next)
-			if (CHECK_FLAG(pi->flags, BGP_PATH_SELECTED) &&
-			    pi->type == ZEBRA_ROUTE_BGP)
```

Comment:
the tests in the new function doesn't look the same - that seems ... questionable?

Reply:
I added a parameter to handle this case.

Reply:
What about pi->subtype and BGP_ROUTE_NORMAL, the new code tests for this as well, when before it doesn't.   This looks like it would break STATIC/AGGREGATES/REDISTRIBUTION here right?